be/src/common/status.cpp

```diff
@@ -181,23 +193,50 @@ Slice Status::message() const {
         return Slice();
     }
 
-    uint32_t length;
-    memcpy(&length, _state, sizeof(length));
-    return Slice(_state + 7, length);
+    uint16_t len1;
+    memcpy(&len1, _state, sizeof(len1));
+    return Slice(_state + 5, len1);
 }
 
+Slice Status::detailed_message() const {
+    if (_state == nullptr) {
+        return Slice();
+    }
+
+    uint16_t len1;
+    uint16_t len2;
+    memcpy(&len1, _state + 0, sizeof(len1));
+    memcpy(&len2, _state + 2, sizeof(len2));
+    uint32_t length = static_cast<uint32_t>(len1) + len2;
+    return Slice(_state + 5, length);
+}
 Status Status::clone_and_prepend(const Slice& msg) const {
     if (ok()) {
         return *this;
     }
-    return Status(code(), msg, precise_code(), message());
+    auto msg2 = message();
+    std::string_view msg_view(reinterpret_cast<const char*>(msg.data), msg.size);
+    std::string_view msg_view2(reinterpret_cast<const char*>(msg2.data), msg2.size);
+    return Status(code(), fmt::format("{}:{}", msg_view, msg_view2));
 }
 
 Status Status::clone_and_append(const Slice& msg) const {
     if (ok()) {
         return *this;
     }
-    return Status(code(), message(), precise_code(), msg);
+    auto msg2 = message();
+    std::string_view msg_view(reinterpret_cast<const char*>(msg.data), msg.size);
+    std::string_view msg_view2(reinterpret_cast<const char*>(msg2.data), msg2.size);
+    return Status(code(), fmt::format("{}:{}", msg_view2, msg_view));
+}
+
+Status Status::clone_and_append_context_line(const char* filename, int line, const char* expr) const {
```

Comment:
I think it's a stack trace. Why not print only the stack trace in the error place directly. It can reduce the memory copy.

Reply:
- Not all places of RETURN_IF_ERROR need an error log
- If you do so, a single error may produce multiple logs, which is annoying and distracting
- This is not a hot path